src/content/components/NodeIcon.js

```diff
@@ -0,0 +1,54 @@
+import React, { PureComponent, PropTypes } from 'react';
+import DefaultFavicon from '../../../res/default-favicon.svg';
+
+const failedIcons = new Set();
```

Comment:
global-ish data. It's a cache that will never be invalidated over the course of a run, and that's clearly bound, as we can't have more members than hostnames in the profile, so I think it's fine.